src/promptflow/promptflow/_sdk/entities/_flow.py

```diff
@@ -142,37 +156,39 @@ def path(self) -> Path:
             )
         return flow_file
 
-    @property
-    def context(self) -> FlowContext:
-        return self._context
-
-    @context.setter
-    def context(self, val):
-        if not isinstance(val, FlowContext):
-            raise UserErrorException("context must be a FlowContext object, got {type(val)} instead.")
-        self._context = val
+    @classmethod
+    def _is_dag_flow(cls, data: dict):
+        """Check if the flow is an DAG flow. Use field 'nodes' to determine."""
+        return data.get(NODES)
 
     @classmethod
     def load(
         cls,
         source: Union[str, PathLike],
         **kwargs,
     ):
+        from promptflow._sdk.entities._eager_flow import EagerFlow
+
         source_path = Path(source)
         if not source_path.exists():
             raise UserErrorException(f"Source {source_path.absolute().as_posix()} does not exist")
-
         flow_path = resolve_flow_path(source_path)
-        if flow_path.exists():
-            # TODO: for file, we should read the yaml to get code and set path to source_path
+        if not flow_path.exists():
+            raise UserErrorException(f"Flow file {flow_path.absolute().as_posix()} does not exist")
+        if flow_path.suffix in [".yaml", ".yml"]:
             # read flow file to get hash
             with open(flow_path, "r", encoding=DEFAULT_ENCODING) as f:
                 flow_content = f.read()
-                flow_dag = yaml.safe_load(flow_content)
+                data = yaml.safe_load(flow_content)
                 kwargs["content_hash"] = hash(flow_content)
-            return cls(code=flow_path.parent.absolute().as_posix(), dag=flow_dag, **kwargs)
-
-        raise UserErrorException("Source must be a directory or a 'flow.dag.yaml' file")
+            is_dag_flow = cls._is_dag_flow(data)
+            # TODO: schema validation and warning on unknown fields
```

Comment:
@wangchao1230 I remember we'll only print warning on unknown fields, right?

Reply:
Yes. Only for unkown fields.